src/Workspaces/CSharp/Portable/Extensions/DefaultExpressionSyntaxExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Utilities;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions
+{
+    internal static class DefaultExpressionSyntaxExtensions
+    {
+        private static readonly LiteralExpressionSyntax s_defaultLiteralExpression = SyntaxFactory.LiteralExpression(SyntaxKind.DefaultLiteralExpression);
+
+        public static bool CanReplaceWithDefaultLiteral(
+            this DefaultExpressionSyntax defaultExpression,
+            CSharpParseOptions parseOptions,
+            OptionSet options,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (parseOptions.LanguageVersion < LanguageVersion.CSharp7_1 ||
+                !options.GetOption(CSharpCodeStyleOptions.PreferSimpleDefaultExpression).Value)
+            {
+                return false;
+            }
+
+            // Using the speculation analyzer can be slow.  Check for common cases first before
+            // trying the expensive path.
+            return CanReplaceWithDefaultLiteralFast(defaultExpression, semanticModel, cancellationToken) ??
+                   CanReplaceWithDefaultLiteralSlow(defaultExpression, semanticModel, cancellationToken);
+        }
+
+        private static bool? CanReplaceWithDefaultLiteralFast(DefaultExpressionSyntax defaultExpression, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (defaultExpression.IsParentKind(SyntaxKind.EqualsValueClause))
+            {
+                var equalsValueClause = (EqualsValueClauseSyntax)defaultExpression.Parent;
+                var typeSyntax = GetTypeSyntax(equalsValueClause);
+
+                if (typeSyntax != null)
+                {
+                    if (typeSyntax.IsVar)
```

Comment:
I'm OK with leaving this as is, but don't we have a helper for determining whether a `TypeSyntax` is *really* var?